### PR TITLE
MULE-14643: Review documentation for error handling - missing parts

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: mule-user-guide/v/4.0/error-handling.adoc [hunk 1]
```diff
@@ -1,22 +1,58 @@
...
 :keywords: error handling, exceptions, exception catching, exceptions
 
-From a high level perspective, errors that occur in Mule fall into one of two categories: system errors, and messaging errors.
+From a high level perspective, errors that occur in Mule fall into one of two categories: system errors and messaging errors.
 
 == Messaging Errors
 
-Mule throws a messaging error whenever an exception is thrown within a flow. This error is known as a Mule Error. +
-These errors are handled by the error handler component. The error handler can enclose On-Error scopes that you can implement to handle errors in your flow. All Mule applications have a default Error Handler for errors not defined in other On Error scopes.
+Mule throws a messaging error whenever an exception is thrown within a flow. This error is known as a link:mule-user-guide/v/4.0/mule-error-concept[Mule Error]. +
```
Comment (pablolagreca) [inline mule-user-guide/v/4.0/error-handling.adoc]: "Mule throws a messaging error whenever an exception is thrown within a flow" -> I would rephrase without talking about exceptions.

File: mule-user-guide/v/4.0/mule-error-concept.adoc [hunk 2]
```diff
@@ -38,24 +38,63 @@ Cause: a ResponseValidatorTypedException instance
...
 
-These hierarchies allow routing to be more general. For example, a handler for MULE:SECURITY catches HTTP unauthorized errors as well as OAuth errors. +
+These hierarchies allow routing to be more general. For example, a handler for `MULE:SECURITY` catches HTTP unauthorized errors as well as OAuth errors. +
 Below you can see what the core runtime’s hierarchy looks like:
 
 image:error-hierarchy.png[Error Hierarchy]
 
-All errors are either general or CRITICAL, the latter being so severe that they cannot be handled. At the top of the general hierarchy is ANY, which allows matching all types under it. It’s important to note the UNKNOWN type, which is used when no clear reason for the failure is found.
+All errors are either general or `CRITICAL`, the latter being so severe that they cannot be handled. At the top of the general hierarchy is `ANY`, which allows matching all types under it. It’s important to note the `UNKNOWN` type, which is used when no clear reason for the failure is found.
```
Comment (pablolagreca) [inline mule-user-guide/v/4.0/mule-error-concept.adoc]: Do you explain somewhere else what mule does when there is a CRITICAL error?

File: mule-user-guide/v/4.0/on-error-scope-concept.adoc [hunk 3]
```diff
@@ -34,20 +34,43 @@ image:error-handling-e77ec.png[]
 If a transaction is being handled, it would be committed as well.
 |===
 
-Each On Error Scope can be set up to be triggered by a different set of error types. Both in Studio and Design Center, you can pick from a list of errors that could be raised by the components used in the error handler's owner.
+Each On Error Scope can be set up to accept a different set of error types. Both in Studio and Design Center, you can pick from a list of errors that could be raised by the components used in the error handler's owner.
 
-You can also set up a path to be triggered by type *ANY*, referring to all error types that aren't used by other paths. Note that conditions are evaluated in order, so for other paths to be evaluated, the *ANY* path must always be the last.
+You can also set up a path that accepts the type *ANY*, matching all error types that aren't used by other paths. Note that conditions are evaluated in order, so for other paths to be evaluated, the *ANY* path must always be the last.
+Keep in mind that without a explicit final path for *ANY*, default error handling will apply and the error will be logged and propagated.
```
Comment (pablolagreca) [inline mule-user-guide/v/4.0/on-error-scope-concept.adoc]: Have you defined the "propagated" concept somewhere? Or perhaps add clarification below with a Note or something like that?

**Comment by pablolagreca** [comment]
Is raise-error documented somewhere else?

**Comment by afelisatti** [comment]
Yes, the raise error docs have been approved by MG and now await being merged.
